### PR TITLE
feat(@angular/cli): autodetect default package manager

### PR DESCRIPTION
**Pull request by clydin**

### CODE REVIEW

File: packages/angular/cli/utilities/check-package-manager.ts [hunk 1]
```diff
@@ -5,49 +5,54 @@
...
-          }
-        }
-      });
 }
 
-function checkYarn() {
-  return execPromise('yarn --version')
-    .then(() => true, () => false);
-}
+export function getPackageManager(root: string): string {
```
Comment (filipesilva) [inline packages/angular/cli/utilities/check-package-manager.ts]: Filename suggests the main export is `checkPackageManager` but instead it is `getPackageManager`. One of these should probably change. From the code, it looks like `getPackageManager` is the right name.

File: packages/angular/cli/utilities/config.ts [hunk 2]
```diff
@@ -151,7 +151,7 @@ export function getProjectByCwd(workspace: experimental.workspace.Workspace): st
   }
 }
 
-export function getPackageManager(): string {
+export function getPackageManager(): string | null {
```
Comment (filipesilva) [inline packages/angular/cli/utilities/config.ts]: Maybe this should change to `getConfiguredPackageManager ` to avoid confusion?

Comment (clydin) [reply packages/angular/cli/utilities/config.ts]: Done.

File: packages/angular/cli/utilities/check-package-manager.ts (hunk 1 above)
Comment (clydin) [reply packages/angular/cli/utilities/check-package-manager.ts]: Done.

**Pull request merged**